Mods/src/PuppiMod.cc

```diff
@@ -141,235 +160,222 @@ void PuppiMod::SlaveBegin()
-    if (numCHPU[i0] == numFCHPUis0[i0])
-      alphaFMed[i0] = 0;
-    else if ((numCHPU[i0] - numFCHPUis0[i0]) % 2 == 0) 
-      alphaFMed[i0] = (alphaFCHPU[i0][IndicesFCHPU[i0][medIndexF - 1]] + alphaFCHPU[i0][IndicesFCHPU[i0][medIndexF]])/2;
-    else 
-      alphaFMed[i0] = alphaFCHPU[i0][IndicesFCHPU[i0][medIndexF]];
-    if (numCHPU[i0] == numCCHPUis0[i0])
-      alphaCMed[i0] = 0;
-    else if ((numCHPU[i0] - numCCHPUis0[i0]) % 2 == 0) 
-      alphaCMed[i0] = (alphaCCHPU[i0][IndicesCCHPU[i0][medIndexC - 1]] + alphaCCHPU[i0][IndicesCCHPU[i0][medIndexC]])/2;
-    else 
-      alphaCMed[i0] = alphaCCHPU[i0][IndicesCCHPU[i0][medIndexC]];
-    
+  std::vector<Double_t> alphaFMed(fNumEtaBins);
+  std::vector<Double_t> alphaCMed(fNumEtaBins);
+  std::vector<Double_t> sigma2F(fNumEtaBins, 0.);
+  std::vector<Double_t> sigma2C(fNumEtaBins, 0.);
+
+  for (UInt_t iEtaBin = 0; iEtaBin < fNumEtaBins; iEtaBin++) {
+    Int_t medIndexF = (numCHPU[iEtaBin] + numFCHPUis0[iEtaBin])/2;              // These are sort of meta
+    Int_t medIndexC = (numCHPU[iEtaBin] + numCCHPUis0[iEtaBin])/2;              // Just watch how they are used...
+    if (fDumpingPuppi)
+      std::cout << "In bin " << iEtaBin << " using " << medIndexC << "; " << medIndexF << std::endl;
+
+    if (numCHPU[iEtaBin] == numFCHPUis0[iEtaBin])
+      alphaFMed[iEtaBin] = 0;
+    else if ((numCHPU[iEtaBin] - numFCHPUis0[iEtaBin]) % 2 == 0)
+      alphaFMed[iEtaBin] = (alphaFCHPU[iEtaBin][medIndexF - 1] + alphaFCHPU[iEtaBin][medIndexF])/2;
+    else
+      alphaFMed[iEtaBin] = alphaFCHPU[iEtaBin][medIndexF];
+
+    if (numCHPU[iEtaBin] == numCCHPUis0[iEtaBin])
+      alphaCMed[iEtaBin] = 0;
+    else if ((numCHPU[iEtaBin] - numCCHPUis0[iEtaBin]) % 2 == 0)
+      alphaCMed[iEtaBin] = (alphaCCHPU[iEtaBin][medIndexC - 1] + alphaCCHPU[iEtaBin][medIndexC])/2;
+    else
+      alphaCMed[iEtaBin] = alphaCCHPU[iEtaBin][medIndexC];
+
     // Now compute the sigma2s
-    sigma2F[i0] = 0;
-    sigma2C[i0] = 0;
-    for (Int_t i1 = numFCHPUis0[i0]; i1 < medIndexF; i1++) 
-      sigma2F[i0] = sigma2F[i0] + pow((alphaFMed[i0]-alphaFCHPU[i0][IndicesFCHPU[i0][i1]]),2);
-    sigma2F[i0] = sigma2F[i0]/(medIndexF - numFCHPUis0[i0]);
-    for (Int_t i1 = numCCHPUis0[i0]; i1 < medIndexC; i1++) 
-      sigma2C[i0] = sigma2C[i0] + pow((alphaCMed[i0]-alphaCCHPU[i0][IndicesCCHPU[i0][i1]]),2);
-    sigma2C[i0] = sigma2C[i0]/(medIndexC - numCCHPUis0[i0]);
-    
-    alphaFMed[i0] = alphaFMed[i0] * (fMedEtaSFs[i0]);                 // Scale the medians
-    alphaCMed[i0] = alphaCMed[i0] * (fMedEtaSFs[i0]);
-
-    sigma2F[i0] = sigma2F[i0] * (fRMSScaleFactor) * (fRMSEtaSFs[i0]); // Scale the sigmas
-    sigma2C[i0] = sigma2C[i0] * (fRMSScaleFactor) * (fRMSEtaSFs[i0]);
+    for (Int_t iAlpha = numFCHPUis0[iEtaBin]; iAlpha < medIndexF; iAlpha++)
+      sigma2F[iEtaBin] += pow((alphaFMed[iEtaBin]-alphaFCHPU[iEtaBin][iAlpha]),2);
+
+    // FIXME medIndexF can be == numFCHPUis0[iEtaBin]
```

Comment:
This has to be solved ASAP. What should sigma2F be when medIndexF == numFCHPUis0?